rnum/src/function/mod.rs

```diff
@@ -2,48 +2,194 @@ use std::sync::Arc;
+/// use rnum::function::std_ops::ln::NaturalLog;
+///
+/// // define "x" as a symbol
+/// let x = Function::Variable("x");
+///
+/// // define 3ln(2x) with a rust method style
+/// let g = 3.0 * (2.0 * &x).ln();
+///
+/// // define 3sin(2x) with rust function style
+/// let f = 3.0 * Function::sin(2.0 * &x);
+/// ```
+///
+/// Please note that the rust variable names have no impact on the function symbols.
+/// However, it is by convention that you name variables according to their symbols
+/// to improve readability.
+/// ```
+/// use rnum::function::Function;
+///
+/// // define a variable "x" according to convention
+/// let x = Function::Variable("x");
+/// // define x^2
+/// let f_x = &x * &x;
+///
+/// // define a variable "x" called y (like a psychopath; don't do this)
+/// let y = Function::Variable("x");
+/// // define x^2
+/// let g_x = &y * &y;
+///
+/// // both functions are the same despite the tomfoolery afoot
+/// assert_eq!(f_x, g_x);
+/// // so are both variables
+/// assert_eq!(x, y);
+/// ```
+/// ## Macro
+/// This a compile time definition method for functions.
+/// It will support a subset of tex.
+/// That will be defined in the currently unnamed syntax format spec.
+/// (we are looking for a name rn lol).
+///
+/// The macro is designed to be as natural as possible while limited to ascii.
+/// ```
+/// use rnum_macros::function;
+///
+/// // define a function with the macro
+/// let f = function!{
+///     d/dx 3 ln(x + 5)
+/// };
+///
+/// assert_eq!(f, function!{3 / (x + 5)});
+/// ```
+/// The function simplified as much as possible
+///
+/// See the specific documentation for the language spec and the macro for more information.
+///
+/// ## Parser
+/// The parser is the backend that powers the `function!` macro.
+/// It consumes whatever we decided to call our syntax.
+/// It can be called directly if you would like to parse syntax at runtime.
+/// This is useful for taking dynamic user input,
+/// for example, if you want to build rnum into a calculator application.
+/// See the parser documentation for more detail.
+///
+///
 #[derive(Clone, Debug)]
 #[allow(non_camel_case_types)]
 pub enum Function<'f> {
+    /// Represents a variable with a &str symbol
     Variable(&'f str),
+    /// A single constant number. Wraps a f64
     Constant(f64),
+    /// Internally used, is marked to be removed when possible
     Simplified {
         function: Arc<Function<'f>>,
         mul: f64,
         pow: f64,
     },
+    /// [Addition](https://en.wikipedia.org/wiki/Addition)
     Add {
         lhs: Arc<Function<'f>>,
         rhs: Arc<Function<'f>>,
     },
+    /// [Subtraction](https://en.wikipedia.org/wiki/Subtraction)
     Sub {
         lhs: Arc<Function<'f>>,
         rhs: Arc<Function<'f>>,
     },
+    /// [Multiplication](https://en.wikipedia.org/wiki/Multiplication)
     Mul {
         lhs: Arc<Function<'f>>,
         rhs: Arc<Function<'f>>,
     },
+    /// [Division](https://en.wikipedia.org/wiki/Division)
```

Comment:
Division leads to a few articles. I'd recommend either using https://en.wikipedia.org/wiki/Division_(mathematics) or https://en.wikipedia.org/wiki/Division_algorithm, probably the first

Reply:
I will change that!